components/builder-web/app/actions/packages.ts

```diff
@@ -68,12 +78,38 @@ export function fetchPackage(pkg) {
         dispatch(clearPackages());
         depotApi.get(pkg.ident).then(response => {
             dispatch(setCurrentPackage(response["results"]));
+            dispatch(fetchCurrentPackageChannels(pkg));
         }).catch(error => {
             dispatch(setCurrentPackage(undefined, error));
         });
     };
 }
 
+export function fetchCurrentPackageChannels(pkg) {
+    return dispatch => {
+        dispatch(clearPackageChannels());
+        depotApi.getPackageChannels(pkg.ident)
+            .then(channels => dispatch(setCurrentPackageChannels(channels)))
+            .catch(error => dispatch(setCurrentPackageChannels(undefined, error)));
+    };
+}
+
+export function fetchVisiblePackageChannels(pkgs) {
+    return dispatch => {
+        pkgs.forEach((pkg) => {
+
+            // Only fetch for fully qualified packages
+            if (pkg.origin && pkg.name && pkg.version && pkg.release) {
+                depotApi.getPackageChannels(pkg)
+                    .then(channels => {
+                        dispatch(setVisiblePackageChannels(pkg, channels));
+                    })
+                .catch(error => console.error(error));
+            }
+        });
+    };
+}
```

Comment:
And this is what the end result looks like:

![](https://dl.dropboxusercontent.com/s/uhkcccn4uon0my7/2017-07-11%20at%205.33%20PM.png)